### PR TITLE
CWP-2440

### PR DESCRIPTION
**Pull request by mvecha**
Update Scan status API integration in Jenkins

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java [hunk 1]
```diff
@@ -230,6 +218,47 @@ private String getConfigXml(Run build) {
...
+      listener
+              .getLogger()
+              .println("Failed to form scan update request due to - " + e.getMessage());
+    }
+    Response<ScanResponse> response = cwpService.updateScanStatus(scanId,request).execute();
+    return getScanResponse(response);
+  }
+
+  private String getScanResponse(Response<ScanResponse> response) throws IOException {
+    if (response.code() == 200) {
```
Comment (AparnaSaripaka) [inline src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java:244]: Best practice is to compare with http status code .

File: src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java [hunk 2]
```diff
@@ -230,6 +218,47 @@ private String getConfigXml(Run build) {
...
+    UpdateScanRequest request = new UpdateScanRequest();
+    try {
+      request.setStatus(status);
+      request.setStep(step);
+      JSONObject jsonObject = new JSONObject(results);
+      JSONObject resultsBlock = jsonObject.getJSONObject("results");
+      request.setViolationsCount(resultsBlock.optJSONArray("violations").length());
+      request.setPassedCount(resultsBlock.optJSONArray("passed_rules").length());
+      request.setSkippedCount(resultsBlock.optJSONArray("skipped_violations").length());
+    }catch(Exception e){
```
Comment (AparnaSaripaka) [inline src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java:234]: Looks good if code is reformatted.

**Review by AparnaSaripaka — DISMISSED** [review]

File: src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java (hunk 2 above)
Comment (mvecha) [reply src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java]: Formatted and uploaded the change

File: src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java (hunk 1 above)
Comment (mvecha) [reply src/main/java/io/jenkins/plugins/zscaler/ZscalerScan.java]: Changes done and uploaded.

**Review by AparnaSaripaka — DISMISSED** [review]

**Review by ChandarGadi — DISMISSED** [review]

**Pull request closed without merging**